File: Arrays_String/LeetCode/Check_If_N_and_Its_Double_Exist.go
func checkIfExist(arr []int) bool {

	ls := make(map[float64]struct{})
	for _, value := range arr {
		a := float64(float64(value) / 2.0)
		if _, ok := ls[a]; ok {
			return true
		}
		if _, ok := ls[float64(value*2)]; ok {
			return true
		}
		ls[float64(value)] = struct{}{}
	}
	return false
}

File: Arrays_String/LeetCode/Check_If_N_and_Its_Double_Exist.cs
using System.Collections.Generic;

namespace Arrays_String.LeetCode
{
    public class Check_If_N_and_Its_Double_Exist
    {
    public bool CheckIfExist(int[] arr) {
            var hash=new HashSet<double>();
            foreach (var item in arr)
            {
                if(hash.Contains(item/2.0) || hash.Contains(item*2) )
                    return true;
                    hash.Add(item);
                
            }
            return false;
    }
    }
}